### PR TITLE
Authentication and device

### PR DESCRIPTION
**Pull request by karanJ2212**
Hello Reviewer,
This is @karanJ2212  with my coding partners , and this milestone fulfills the below requirements:

Project requirements

 Remove current_user method in ApplicationController, devise will provide us with one.
 [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
 Can register a new user.
 User logs in with a combination of email and password.
 Hashed password should be stored in the database.
 Ask for confirmation of email.
 Can reset password.
 [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @karanJ2212 and @Kaghenimbale,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### The live demo link on your menu header is not working, kindly remove it. Kindly ensure you have no broken links on the readme documentation.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (Zuhaib042) [inline app/views/devise/sessions/new.html.erb:27]: - [ ] Good job, everything is working fine but  your login page needs to match the wireframe design and styling.  so kindly consider style it according to the wireframe.
blog_login

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :bio %><br />
+    <%= f.text_area :bio, autofocus: true %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
```
Comment (Zuhaib042) [inline app/views/devise/registrations/new.html.erb]: - [ ] Good job, everything is working fine but your sign up page needs to match the wireframe design and styling. I'd appreciate your consideration of styling it according to the following wireframe. although this given picture is for the login page, your signup page should also be similar.
blog_login

**Review by Zuhaib042 — CHANGES_REQUESTED** [review]
Hi @karanJ2212 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Perfect implementation
✅ Good styling

## Required Changes ♻️


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KanzaTahreem — APPROVED** [review]
# Approved 🙌

Hi @karanJ2212 

Great work implementing the suggested changes 💪 **_Your project is COMPLETE! There is nothing else to say other than... it's time to merge it! 🚀 Congratulations!_**



## Highlights 🎯
There are some excellent points that you have made perfectly. Keep up the great work! 👏

-  ✅ Styling is looking great, perfect!
-  ✅ Great work using the right GitFlow 
-  ✅ Your README looks good, well done!
-  ✅ Linters are working properly, great job! 

## Optional suggestions ❕
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**